components/SearchBar/index.tsx

```diff
@@ -244,90 +308,53 @@ function Component() {
       verified: false,
       hideWallet: true,
     });
-    if (
-      isValidUsername(thisName) ||
-      //@todo move the following exceptions to isValidUsername in utils
-      thisName === "init" ||
-      thisName === "tyron" ||
-      thisName === "donate" ||
-      thisName === "wfp"
-    ) {
-      updateUser({
-        name: thisName,
-        domain: thisDomain
-      })
-      switch (thisDomain) {
-        case DOMAINS.TYRON:
-          if (VALID_SMART_CONTRACTS.includes(user?.name!))
-            window.open(
-              SMART_CONTRACTS_URLS[
-              user?.name as unknown as keyof typeof SMART_CONTRACTS_URLS
-              ]
-            );
-          else
-            toast('Invalid smart contract.', {
-              position: "top-left",
-              autoClose: 2000,
-              hideProgressBar: false,
-              closeOnClick: true,
-              pauseOnHover: true,
-              draggable: true,
-              progress: undefined,
-              theme: 'dark',
-            });
-          break;
-        case DOMAINS.DID:
-          await resolveDid();
-          break;
-        case DOMAINS.VC:
-          await resolveDomain();
-          break;
-        case DOMAINS.TREASURY:
-          await resolveDomain();
-          break;
-        case DOMAINS.PSC:
-          toast('Coming soon!', {
-            position: "top-left",
-            autoClose: 2000,
-            hideProgressBar: false,
-            closeOnClick: true,
-            pauseOnHover: true,
-            draggable: true,
-            progress: undefined,
-            theme: 'dark',
-          }); //await resolveDomain();
-          break;
-        case DOMAINS.DEX:
-          await resolveDomain();
-          break;
-        case DOMAINS.STAKE:
-          await resolveDomain();
-          break;
-        default:
-          toast('Invalid domain.', {
-            position: "top-left",
-            autoClose: 2000,
-            hideProgressBar: false,
-            closeOnClick: true,
-            pauseOnHover: true,
-            draggable: true,
-            progress: undefined,
-            theme: 'dark',
-          });
-      }
-    } else {
-      setTimeout(() => {
-        toast.error("Invalid username. Names with less than seven characters are premium and will be for sale later on.", {
+    
+    const path = window.location.pathname.replace("/", "").toLowerCase();
+    setSearch(`${setUsername()}.${setDomain()}`)
+    updateUser({
+      name: setUsername(),
+      domain: setDomain()
+    });
+    switch (setDomain()) {
+      case DOMAINS.TYRON:
+        if (VALID_SMART_CONTRACTS.includes(username))
+          window.open(
+            SMART_CONTRACTS_URLS[
+            username as unknown as keyof typeof SMART_CONTRACTS_URLS
+            ]
+          );
+        else setError("Invalid smart contract");
```

Comment:
Let's deprecate the use of setError for a toast notification.

Reply:
okay